### PR TITLE
Breadboard schema package

### PR DESCRIPTION
**Pull request by Mearman**

### CODE REVIEW

File: .changeset/cool-waves-pretend.md
Comment (aomarks) [inline .changeset/cool-waves-pretend.md:2]: should be minor, since removing a file is breaking

File: .changeset/cool-waves-pretend.md
Comment (aomarks) [inline .changeset/cool-waves-pretend.md:5]: s/bradboard/breadboard :)

File: .changeset/four-bottles-march.md
Comment (aomarks) [inline .changeset/four-bottles-march.md:1]: remove this changeset I think? seems redundant given the other one

File: packages/schema/prepack.ts
Comment (aomarks) [inline packages/schema/prepack.ts:1]: Needs copyright header

File: packages/schema/package.json
Comment (aomarks) [inline packages/schema/package.json:9]: I think we can just rely on the top-level wireit dependency. Otherwise when we bump versions we'll get duplicates. Same for any other shared deps (e.g. types/node?)

File: packages/schema/package.json
Comment (aomarks) [inline packages/schema/package.json:22]: Smart!

File: packages/schema/prepack.ts
Comment (aomarks) [inline packages/schema/prepack.ts:1]: Mind putting these two `.ts` files in `scripts/` ? I think that's a good convention for "stuff that's just for working on this package".

File: packages/schema/board.json
Comment (aomarks) [inline packages/schema/board.json:1]: Maybe put this in `scripts/testdata`?

**Review by aomarks — COMMENTED** [review]
Nice!

**Review by aomarks — CHANGES_REQUESTED** [review]

File: .changeset/cool-waves-pretend.md
Comment (Mearman) [reply .changeset/cool-waves-pretend.md:2]:
```suggestion
"@google-labs/breadboard": minor
```

File: packages/schema/validate.ts
Comment (Mearman) [reply packages/schema/validate.ts:1]:
```suggestion
/**
 * @license
 * Copyright 2023 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import addFormats from "ajv-formats";
```

File: packages/schema/prepack.ts
Comment (Mearman) [reply packages/schema/prepack.ts:1]:
```suggestion
/**
 * @license
 * Copyright 2023 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import fs from "fs";
```

File: packages/schema/package.json
Comment (Mearman) [reply packages/schema/package.json:22]: which bit?

File: packages/schema/package.json
Comment (Mearman) [reply packages/schema/package.json:9]: thoughts on using something like syncpack?

File: CODEOWNERS
Comment (aomarks) [inline CODEOWNERS:1]: Not sure we want this. It means that you have to approve any change we make to schema. I value your opinion very much, but I'm not sure it should be blocking :)

File: packages/schema/package.json
Comment (aomarks) [reply packages/schema/package.json:19]: Oh just using prepack to enforce that it validates so that it's impossible to publish  with validation errors.

File: packages/schema/scripts/prepack.ts
Comment (aomarks) [inline packages/schema/scripts/prepack.ts:3]: 2024

File: packages/schema/scripts/validate.ts
Comment (aomarks) [inline packages/schema/scripts/validate.ts:3]: 2024

File: packages/schema/scripts/prepack.ts
Comment (aomarks) [inline packages/schema/scripts/prepack.ts:9]: Whoa, what is `~` in an import path?

**Review by aomarks — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (Mearman) [reply CODEOWNERS:1]: Oh I thought on GitHub by default it just pinged the owner on touching PRs will read the docs more

File: packages/schema/scripts/prepack.ts
Comment (Mearman) [reply packages/schema/scripts/prepack.ts:9]: haha just a path alias
https://github.com/breadboard-ai/breadboard/pull/538/files/6e744544fc382858ea8e62ab5570023289e3aea7#diff-fc763b82b8113516d39db871392775c27b0e8427331bd7f0add8c64d3101802bR6-R8

File: packages/schema/scripts/validate.ts
Comment (Mearman) [reply packages/schema/scripts/validate.ts:3]:
```suggestion
 * Copyright 2024 Google LLC
```
still getting used to it

File: packages/schema/scripts/prepack.ts
Comment (Mearman) [reply packages/schema/scripts/prepack.ts:3]:
```suggestion
 * Copyright 2024 Google LLC
```

File: packages/schema/package.json
Comment (Mearman) [reply packages/schema/package.json:19]: I'm always a fun of hooking into lifecycle stages. especially if it means not having to install git hooks

File: CODEOWNERS
Comment (Mearman) [reply CODEOWNERS:1]: Yeah so it's an optional addition to require codeowner signoff

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#:~:text=When%20someone%20with%20admin%20or%20owner%20permissions%20has%20enabled%20required%20reviews%2C%20they%20also%20can%20optionally%20require%20approval%20from%20a%20code%20owner%20before%20the%20author%20can%20merge%20a%20pull%20request%20in%20the%20repository.%20For%20more%20information%2C%20see%20%22About%20protected%20branches.%22

Comment (Mearman) [reply CODEOWNERS:1]: Though apparently, it only works with write permission so I shall remove

File: packages/schema/scripts/prepack.ts
Comment (aomarks) [reply packages/schema/scripts/prepack.ts:9]: Huh. What happens when you `tsc compile` this? Does tsc rewrite the specifier to something?

Why not just do `from "../breadboard.schema.json"`?

Comment (Mearman) [reply packages/schema/scripts/prepack.ts:9]: tsc leaves paths alone. with a bundler they would be rewritten. I prefer it for two reasons. I think the absolute import is tidier. And no need to refactor absolute imports when moving a file.

Comment (aomarks) [reply packages/schema/scripts/prepack.ts:9]: I prefer the relative import. It will work without ts-node and is the normal idiomatic way to refer to other files. I would rate those concerns higher than the others.

Comment (Mearman) [reply packages/schema/scripts/prepack.ts:9]: reasonable. will update

**Review by aomarks — APPROVED** [review]

**Pull request merged**
